tortoisebeacon/tortoisebeacon.go

```diff
@@ -182,94 +186,39 @@ func (tb *TortoiseBeacon) Close() error {
 	return nil
 }
 
-// Get returns a Tortoise Beacon value as types.Hash32 for a certain epoch or an error if it doesn't exist.
-// TODO(nkryuchkov): Consider making unexported.
-func (tb *TortoiseBeacon) Get(epochID types.EpochID) (types.Hash32, error) {
+// Get returns a Tortoise Beacon value as []byte for a certain epoch or an error if it doesn't exist.
+func (tb *TortoiseBeacon) Get(layerID types.LayerID) ([]byte, error) {
+	epochID := layerID.GetEpoch()
+
 	if tb.tortoiseBeaconDB != nil {
 		if val, ok := tb.tortoiseBeaconDB.GetTortoiseBeacon(epochID); ok {
-			return val, nil
+			return val.Bytes(), nil
 		}
 	}
 
 	tb.beaconsMu.RLock()
-	beacon, ok := tb.beacons[epochID]
-	tb.beaconsMu.RUnlock()
+	defer tb.beaconsMu.RUnlock()
 
-	if !ok {
-		return types.Hash32{}, ErrBeaconNotCalculated
-	}
+	var beacon types.Hash32
+	var ok bool
 
-	if tb.tortoiseBeaconDB != nil {
-		if err := tb.tortoiseBeaconDB.SetTortoiseBeacon(epochID, beacon); err != nil {
-			return types.Hash32{}, fmt.Errorf("update beacon in DB: %w", err)
+	if tb.beaconAlreadyCalculated(layerID) {
+		if beacon, ok = tb.beacons[epochID]; !ok {
+			return nil, ErrBeaconNotCalculated
+		}
+	} else {
+		if beacon, ok = tb.beacons[epochID-1]; !ok {
+			return nil, ErrBeaconNotCalculated
 		}
 	}
 
-	return beacon, nil
-}
-
-// GetBeacon waits until a Tortoise Beacon value is ready for a certain epoch and returns it as []byte.
-func (tb *TortoiseBeacon) GetBeacon(epochNumber types.EpochID) []byte {
-	if err := tb.Wait(epochNumber); err != nil {
-		tb.Log.With().Error("Failed to wait for tortoise beacon value calculation",
-			log.Uint64("epoch_id", uint64(epochNumber)),
-			log.Err(err))
-
-		return nil
-	}
-
-	v, err := tb.Get(epochNumber)
-	if err != nil {
-		return nil
-	}
-
-	return v.Bytes()
-}
-
-// Wait waits until beacon for this epoch becomes ready.
-func (tb *TortoiseBeacon) Wait(epochID types.EpochID) error {
-	tb.waitUntilStarted()
-
 	if tb.tortoiseBeaconDB != nil {
-		if _, ok := tb.tortoiseBeaconDB.GetTortoiseBeacon(epochID); ok {
-			return nil
+		if err := tb.tortoiseBeaconDB.SetTortoiseBeacon(epochID, beacon); err != nil {
```

Comment:
every time you will get the beacon you will try to insert it to db? won't it cause an error every time since you put it in the first time?

Reply:
If it already exists in a DB for a certain epoch, it's returned from the DB. So, it's put in a DB once every epoch.